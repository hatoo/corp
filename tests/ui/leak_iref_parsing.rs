use futures::FutureExt;
use stap::{Cursor, Input, Parsing};

fn main() {
    let mut input = Input::new(Cursor {
        buf: Vec::<u8>::new(),
        index: 0,
    });

    let mut leak = None;

    let mut p = Parsing::new(&mut input, |iref| {
        let leak_mut = &mut leak;
        async move {
            *leak_mut = Some(iref);
        }
        .boxed_local()
    });

    p.poll();

    drop(leak);
}
