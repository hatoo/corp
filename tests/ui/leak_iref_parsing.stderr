error[E0505]: cannot move out of `leak` because it is borrowed
  --> tests/ui/leak_iref_parsing.rs:22:10
   |
12 |     let mut p = Parsing::new(&mut input, |iref| {
   |                                          ------ borrow of `leak` occurs here
13 |         let leak_mut = &mut leak;
   |                             ---- borrow occurs due to use in closure
...
22 |     drop(leak);
   |          ^^^^ move out of `leak` occurs here
23 | }
   | - borrow might be used here, when `p` is dropped and runs the destructor for type `Parsing<'_, u8, (), Pin<Box<dyn futures::Future<Output = ()>>>>`
